### PR TITLE
video/image_writer: attach MDVC metadata and CLLI metadata

### PR DESCRIPTION
**Pull request by Traneptora**
Note: test with https://patchwork.ffmpeg.org/project/ffmpeg/list/?series=10666

### CODE REVIEW

**Comment by kasper93** [comment]
This probably should also be done in `f_favfi.c:init_pads()` and `mp_image.c.c:mp_image_to_av_frame()` in a way it doesn't duplicate code needless to say.

**Comment by Traneptora** [comment]
Good call

File: video/image_writer.c [hunk 1]
```diff
@@ -148,6 +149,51 @@ static void prepare_avframe(AVFrame *pic, AVCodecContext *avctx,
         pl_system_to_av(image->params.repr.sys);
     avctx->chroma_sample_location = pic->chroma_location =
         pl_chroma_to_av(image->params.chroma_location);
+
+    struct pl_hdr_metadata *meta = &image->params.color.hdr;
+    AVMasteringDisplayMetadata *mdm = NULL;
+    if (meta->min_luma > 0.0f && meta->max_luma > meta->min_luma) {
+        mdm = av_mastering_display_metadata_create_side_data(pic);
+        if (!mdm) {
+            mp_dbg(log, "error creating AVMasteringDisplayMetadata");
```
Comment (sfan5) [inline video/image_writer.c:158]: btw: `MP_HANDLE_OOM`

**Comment by Traneptora** [comment]
Note: `f_lavfi.c -> init_pads` couldn't be changed because AVBufferSrcParams doesn't have the necessary fields to assign.

File: video/image_writer.c [hunk 2]
```diff
@@ -135,19 +135,18 @@ static void prepare_avframe(AVFrame *pic, AVCodecContext *avctx,
     pic->format = avctx->pix_fmt;
     pic->width = avctx->width;
     pic->height = avctx->height;
-    avctx->color_range = pic->color_range =
-        pl_levels_to_av(image->params.repr.levels);
+    pl_avframe_set_repr(pic, image->params.repr);
```
Comment (kasper93) [inline video/image_writer.c:138]: I think you meant `_color`

File: video/mp_image.c [hunk 3]
```diff
@@ -1168,11 +1168,7 @@ struct AVFrame *mp_image_to_av_frame(struct mp_image *src)
     if (src->fields & MP_IMGFIELD_REPEAT_FIRST)
         dst->repeat_pict = 1;
 
-    dst->colorspace = pl_system_to_av(src->params.repr.sys);
-    dst->color_range = pl_levels_to_av(src->params.repr.levels);
-    dst->color_primaries =
-        pl_primaries_to_av(src->params.color.primaries);
-    dst->color_trc = pl_transfer_to_av(src->params.color.transfer);
+    pl_avframe_set_repr(dst, src->params.repr);
```
Comment (kasper93) [inline video/mp_image.c:1171]: I think we need `_color` here too

File: video/image_writer.c (hunk 2 above)
Comment (Traneptora) [reply video/image_writer.c:138]: No, see below. That's guarded behind tag_csp

File: video/mp_image.c (hunk 3 above)
Comment (Traneptora) [reply video/mp_image.c:1171]: That's alreasy called lower down in the function.

File: video/image_writer.c [hunk 4]
```diff
@@ -135,19 +135,18 @@ static void prepare_avframe(AVFrame *pic, AVCodecContext *avctx,
...
 
     if (!tag_csp)
         return;
-    avctx->color_primaries = pic->color_primaries =
-        pl_primaries_to_av(image->params.color.primaries);
-    avctx->color_trc = pic->color_trc =
-        pl_transfer_to_av(image->params.color.transfer);
-    avctx->colorspace = pic->colorspace =
-        pl_system_to_av(image->params.repr.sys);
+    pl_avframe_set_color(pic, image->params.color);
```
Comment (Traneptora) [inline video/image_writer.c:144]: @kasper93 here

File: video/mp_image.c (hunk 3 above)
Comment (Traneptora) [reply video/mp_image.c:1171]: https://github.com/mpv-player/mpv/blob/a45518cf571fd1b43482aa4720eeb7a4b0b1c684/video/mp_image.c#L1191

Comment (kasper93) [reply video/mp_image.c:1171]: If possible please move them closer together, they are linked and is nicer to have them both next to each other

**Review by kasper93 — APPROVED** [review]
Should be fine.

**Pull request merged**